client/validate.go

```diff
@@ -0,0 +1,98 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"golang.org/x/xerrors"
+)
+
+// ValidateApplyOptions carries options for ApplyValidationSet.
+type ValidateApplyOptions struct {
+	Mode  string
+	PinAt int
+}
+
+// ValidationSetResult holds information about a single validation set.
+type ValidationSetResult struct {
+	ValidationSet string `json:"validation-set,omitempty"`
+	Mode          string `json:"mode"`
+	Seq           int    `json:"seq,omitempty"`
+	Valid         bool   `json:"valid"`
+	// TODO: flags/states for notes column
+}
+
+// ApplyValidationSet applies or forgets the given validation set identified by account and name.
+func (client *Client) ApplyValidationSet(account, name string, opts *ValidateApplyOptions) error {
+	q := url.Values{}
+	q.Set("validation-set", fmt.Sprintf("%s/%s", account, name))
```

Comment:
to follow what we do with other entities this two would go in the JSON as well?

Reply:
I think for consistency with GET handling below, this would use "/v2/validation-set/account/name" path instead? and POST json data would carry mode/pin-at?

Reply:
that works as well , with plural though "/v2/validations-sets/..." . For snaps we actually I think support POSTs both to /v2/snaps and /v2/snaps/{name} but is probably overkill here at least initially.